commit bd024f07019f5d9fea236675607a69f74a66bc7b	bd024f07019f5d9fea236675607a69f74a66bc7b
Author: erouault <erouault>
Date:   Mon Aug 15 21:26:56 2016 +0000

    * tools/rgb2ycbcr.c: validate values of -v and -h parameters to
    avoid potential divide by zero. Fixes CVE-2016-3623 (bugzilla #2569)

diff --git a/ChangeLog b/ChangeLog
index 5d606087..3e6642a3 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+2016-08-15 Even Rouault <even.rouault at spatialys.com>
+
+	* tools/rgb2ycbcr.c: validate values of -v and -h parameters to
+	avoid potential divide by zero. Fixes CVE-2016-3623 (bugzilla #2569)
+
 2016-08-15 Even Rouault <even.rouault at spatialys.com>
 
 	* tools/tiffcrop.c: Fix out-of-bounds write in loadImage().
diff --git a/tools/rgb2ycbcr.c b/tools/rgb2ycbcr.c
index 3829d6b2..51f42598 100644
--- a/tools/rgb2ycbcr.c
+++ b/tools/rgb2ycbcr.c
@@ -95,9 +95,13 @@ main(int argc, char* argv[])
 			break;
 		case 'h':
 			horizSubSampling = atoi(optarg);
+            if( horizSubSampling != 1 && horizSubSampling != 2 && horizSubSampling != 4 )
+                usage(-1);
 			break;
 		case 'v':
 			vertSubSampling = atoi(optarg);
+            if( vertSubSampling != 1 && vertSubSampling != 2 && vertSubSampling != 4 )
+                usage(-1);
 			break;
 		case 'r':
 			rowsperstrip = atoi(optarg);
